### PR TITLE
Switch route controller to shared informers

### PR DESCRIPTION
**Pull request by ncdc**
Initially part of #40097 

@smarterclayton @gmarek @wojtek-t @timothysc @kubernetes/sig-scalability-pr-reviews @sttts @deads2k @liggitt

### CODE REVIEW

File: pkg/controller/route/routecontroller.go [hunk 1]
```diff
@@ -50,45 +50,39 @@ const (
...
-
 	return rc
 }
 
-func (rc *RouteController) Run(syncPeriod time.Duration) {
-	go rc.nodeController.Run(wait.NeverStop)
+func (rc *RouteController) Run(stopCh <-chan struct{}, syncPeriod time.Duration) {
+	defer utilruntime.HandleCrash()
+
+	glog.Info("Staring the route controller")
```
Comment (smarterclayton) [inline pkg/controller/route/routecontroller.go]: Typo

Comment (ncdc) [inline pkg/controller/route/routecontroller.go]: Fixed

**Comment by ncdc** [comment]
Re-adding lgtm b/c all I did was fix the typo

**Pull request merged**